### PR TITLE
#168131414 Users should receive signup validation errors

### PR DESCRIPTION
**Pull request by j0flintking02**
#### What does this PR do?
Users should receive signup/Registration validation errors
#### Description of Task to be completed?
- ensure that test case is validated and returns descriptive errors
- implement error handling for failed delivery
#### How should this be manually tested?
- Clone the repo
- checkout to the branch ft-signup-validation-168131414
   and pull the changes
- Run npm install
- add the environment variables to the .env file
- add a secret key in the .env
- Run npm test
- Run npm start for production or npm start: dev for development
- Go to postman and head over to `localhost:3000/api/v1/auth/signup`
- Use the format of the payload as shown below
```json
{
	"userEmail" : "jonathanaurugai@gmail.com",
	"firstName" : "Jonathan",
	"lastName": "Aurugai",
	"userPassword": "Root1234",
	"userRoles":"Travel Team Member"
}
```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168131414](https://www.pivotaltracker.com/story/show/168131414)
#### Screenshots (if appropriate)
Screen Shot 2019-09-11 at 11 57 34

#### Questions:
N/A

### CODE REVIEW

**Review by willywax — APPROVED** [review]
Looks good

**Review by Dkabiswa — COMMENTED** [review]
for try and catch errors it's better to user the error handling middleware we have pass next(error)

**Review by drayzii — CHANGES_REQUESTED** [review]
Can you add descriptive error messages for other fields... The same way you did for the password field

**Review by drayzii — APPROVED** [review]

File: src/validation/userValidator.js [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
+        .max(30)
+        .required(),
+      lastName: Joi.string()
+        .alphanum()
+        .min(3)
+        .max(30)
+        .required(),
+      userPassword: Joi.string()
+        .required()
+        .regex(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z]{8,}$/)
```
Comment (Paulvitalis200) [inline src/validation/userValidator.js]: Add a special characters requirement to strengthen our passwords.

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**